neqo-http3/src/stream_type_reader.rs

```diff
@@ -504,6 +514,20 @@ mod tests {
         );
     }
 
+    #[test]
+    fn decode_grease_stream() {
```

Comment:
I think that it would be better to have a higher-level test here.  Is it possible to add the ability to grease request streams?  That shouldn't be particularly hard.  I took a quick look and `SendMessage::encode()` seems like it could be tweaked to send a grease frame or two.